io/dagreader.go

```diff
@@ -426,10 +427,14 @@ func (dr *dagReader) Seek(offset int64, whence int) (int64, error) {
 					return err
 				}
 
-				_, err = dr.currentNodeData.Seek(left, io.SeekStart)
+				n, err := dr.currentNodeData.Seek(left, io.SeekStart)
 				if err != nil {
 					return err
 				}
+
+				if n != left {
+					return fmt.Errorf("nodeData seeked to wrong position %d, ekpected %d", n, left)
+				}
```

Comment:
Check the comment clarification after the `seek` call, if you're unsatisfied by that reasoning I'm fine with this check.

Reply:
From https://golang.org/pkg/io/#Seeker:
`
Seeking to any positive offset is legal, but the behavior of subsequent I/O operations on the underlying object is implementation-dependent. 
`

Which means that seeking beyond EOF should possible - looking at `bytes.Reader.Seek` as  this is what this check is for - https://golang.org/src/bytes/reader.go?s=2744:2806#L103 - it looks like offset is just passed through without any checks other than checking if it's negative.

(So this check wouldn't even do anything, I'm just going to drop that commit)